packages/core/test/lib/resourceLoader.test.ts.trig

```diff
@@ -10,3 +12,60 @@ graph ex:foo {
 graph ex:bar {
   ex:foo rdf:label "in graph bar".
 }
+
+GRAPH <findResourceLoader/when-resource-shape-has-a-resource-loader> {
+  ex:PersonShape
+    a kopflos:ResourceShape ;
+    kopflos:api ex:PeopleApi ;
+    sh:targetNode ex:Person ;
+    kopflos:resourceLoader ex:PersonResourceLoader ;
+  .
+
+  ex:PeopleApi
+    a kopflos:Api ;
+    kopflos:config ex:Config ;
+    kopflos:resourceLoader ex:ResourceLoader2 ;
+  .
+
+  ex:Config
+    a kopflos:Config ;
+    kopflos:resourceLoader ex:ResourceLoader3 ;
+  .
+}
+
+GRAPH <findResourceLoader/when-resource-shape-has-no-resource-loader> {
+  ex:PersonShape
+    a kopflos:ResourceShape ;
+    kopflos:api ex:PeopleApi ;
+    sh:targetNode ex:Person ;
+  .
+
+  ex:PeopleApi
+    a kopflos:Api ;
+    kopflos:config ex:Config ;
+    kopflos:resourceLoader ex:ApiResourceLoader ;
+  .
+
+  ex:Config
+    a kopflos:Config ;
+    kopflos:resourceLoader ex:ResourceLoader3 ;
+  .
+}
+
+GRAPH <findResourceLoader/when-resource-shape-and-API-have-no-loaders> {
+  ex:PersonShape
+    a kopflos:ResourceShape ;
+    kopflos:api ex:PeopleApi ;
+    sh:targetNode ex:Person ;
+  .
+
+  ex:PeopleApi
+    a kopflos:Api ;
+    kopflos:config ex:Config ;
```

Comment:
I added a new `kl:config` property to avoid the overloaded, bidirectional use of `kl:api`

This way, each level has a property, and the are unidirectional.

`ResourceShape --(kl:api)--> Api --(kl:config)--> Config`